### PR TITLE
Fix warnings about >> and > operator precedence in parser code.

### PR DESCRIPTION
**Pull request by LGM-Doyle**

### CODE REVIEW

**Comment by geoffthemedio** [comment]
Can you fix all the indenting?

**Comment by LGM-Doyle** [comment]
Can you tell me what you want for indenting.  The indenting for the parser code always seems arbitrary.

File: parse/CommonParamsParser.cpp [hunk 1]
```diff
@@ -96,11 +96,11 @@ namespace parse { namespace detail {
                 > (
                         consumable_meter(_r1, _r2)
                     |   consumable_special(_r1, _r2)
-                    |(  '['
+                    |( ('['
                         >> *(
                                 consumable_meter(_r1, _r2)
                             |   consumable_special(_r1, _r2)
-                            )
+                            ))
```
Comment (geoffthemedio) [inline parse/CommonParamsParser.cpp]: the matching `(` and `)` should line up horizontally

File: parse/ConditionParser1.cpp [hunk 2]
```diff
@@ -83,8 +83,8 @@ namespace {
                 ;
 
             owned_by_1
-                =   tok.OwnedBy_
-                >>  parse::detail::label(Empire_token) > parse::int_value_ref()
+                =   (tok.OwnedBy_
```
Comment (geoffthemedio) [inline parse/ConditionParser1.cpp:86]: put the `(` before the `tok` and don't move the `tok`

File: parse/ConditionParser1.cpp [hunk 3]
```diff
@@ -105,9 +105,9 @@ namespace {
                 ;
 
             owned_by_5
-                =   tok.OwnedBy_
+                =   (tok.OwnedBy_
                 >>  parse::detail::label(Affiliation_token) >> parse::empire_affiliation_type_enum() [ _a = _1 ]
-                >>  parse::detail::label(Empire_token)      >  parse::int_value_ref()
+                >>  parse::detail::label(Empire_token)      ) >  parse::int_value_ref()
```
Comment (geoffthemedio) [inline parse/ConditionParser1.cpp]: keep left side of `parse::int_value_ref` aligned with the immediately above `parse`

File: parse/ConditionParser2.cpp [hunk 4]
```diff
@@ -43,8 +43,8 @@ namespace {
                 ;
 
             enqueued1
-                =   (   tok.Enqueued_
-                        >>  parse::detail::label(Type_token)   >>   tok.Building_
+                =   (   (tok.Enqueued_
+                         >>  parse::detail::label(Type_token)   >>   tok.Building_)
```
Comment (geoffthemedio) [inline parse/ConditionParser2.cpp]: keep same-nested level `>` and `>>` left-aligned, particularly when the first thing other than `(` on the line

**Comment by geoffthemedio** [comment]
current code isn't perfect, but basically keep it neat / aligned with similar or related stuff

File: parse/CommonParamsParser.cpp (hunk 1 above)
Comment (LGM-Doyle) [reply parse/CommonParamsParser.cpp]: aligned

File: parse/ConditionParser1.cpp (hunk 2 above)
Comment (LGM-Doyle) [reply parse/ConditionParser1.cpp:86]: aligned.

File: parse/ConditionParser1.cpp [hunk 5]
```diff
@@ -83,8 +83,8 @@ namespace {
                 ;
 
             owned_by_1
-                =   tok.OwnedBy_
-                >>  parse::detail::label(Empire_token) > parse::int_value_ref()
+                =   (tok.OwnedBy_
+                     >>  parse::detail::label(Empire_token)) > parse::int_value_ref()
```
[the remark this answers is not in the document]
Comment (LGM-Doyle) [reply parse/ConditionParser1.cpp]: aligned

File: parse/ConditionParser1.cpp (hunk 3 above)
Comment (LGM-Doyle) [reply parse/ConditionParser1.cpp]: aligned.

File: parse/ConditionParser2.cpp (hunk 4 above)
Comment (LGM-Doyle) [reply parse/ConditionParser2.cpp]: aligned

**Review by LGM-Doyle — COMMENTED** [review]
Thanks for reviewing this.  I think that the matching brackets are now vertically aligned.

File: parse/EffectParser1.cpp [hunk 6]
```diff
@@ -32,15 +32,15 @@ namespace {
             const parse::lexer& tok = parse::lexer::instance();
 
             set_empire_meter_1
-                =    tok.SetEmpireMeter_
-                >>   parse::detail::label(Empire_token) >  parse::int_value_ref() [ _b = _1 ]
+                =    (tok.SetEmpireMeter_
```
Comment (geoffthemedio) [inline parse/EffectParser1.cpp]: ...

File: parse/EffectParser2.cpp [hunk 7]
```diff
@@ -29,17 +29,19 @@ namespace {
             const parse::lexer& tok =                                                       parse::lexer::instance();
 
             set_meter
-                =   parse::set_non_ship_part_meter_type_enum() [ _a = _1 ] /* has some overlap with parse::set_ship_part_meter_type_enum() so can't use '>' */
-                >>  parse::detail::label(Value_token)               > parse::double_value_ref() [ _c = _1 ]
+                =   (parse::set_non_ship_part_meter_type_enum() [ _a = _1 ] /* has some overlap with parse::set_ship_part_meter_type_enum() so can't use '>' */
+                    >>  parse::detail::label(Value_token)
+                    )               > parse::double_value_ref() [ _c = _1 ]
```
Comment (geoffthemedio) [inline parse/EffectParser2.cpp]: ...

File: parse/EffectParser2.cpp [hunk 8]
```diff
@@ -29,17 +29,19 @@ namespace {
...
                     |    eps [ _val = new_<Effect::SetMeter>(_a, _c) ]
                     )
                 ;
 
             set_ship_part_meter
-                =    parse::set_ship_part_meter_type_enum() [ _a = _1 ]
-                >>   parse::detail::label(PartName_token)   > parse::string_value_ref() [ _b = _1 ]
+                =    (parse::set_ship_part_meter_type_enum() [ _a = _1 ]
+                     >>   parse::detail::label(PartName_token)
+                     )   > parse::string_value_ref() [ _b = _1 ]
```
Comment (geoffthemedio) [inline parse/EffectParser2.cpp]: ...

File: parse/EffectParser3.cpp [hunk 9]
```diff
@@ -84,9 +84,9 @@ namespace {
                 ;
 
             add_special_2
-                =  (tok.AddSpecial_ | tok.SetSpecialCapacity_)
+                =  ((tok.AddSpecial_ | tok.SetSpecialCapacity_)
                 >>  parse::detail::label(Name_token) >> parse::string_value_ref() [ _c = _1 ]
-                >> (parse::detail::label(Capacity_token) | parse::detail::label(Value_token))
+                >> (parse::detail::label(Capacity_token) | parse::detail::label(Value_token)))
```
Comment (geoffthemedio) [inline parse/EffectParser3.cpp]: ...

File: parse/EffectParser4.cpp [hunk 10]
```diff
@@ -87,9 +89,10 @@ namespace {
                 ;
 
             create_field_1
-                =   (       tok.CreateField_
-                        >>  parse::detail::label(Type_token)    >>  parse::string_value_ref() [ _a = _1 ]
-                        >>  parse::detail::label(Size_token)    >   parse::double_value_ref() [ _b = _1 ]
+                =   ((       tok.CreateField_
+                     >>  parse::detail::label(Type_token)    >>  parse::string_value_ref() [ _a = _1 ]
+                     >>  parse::detail::label(Size_token)
+                     ) >   parse::double_value_ref() [ _b = _1 ]
```
Comment (geoffthemedio) [inline parse/EffectParser4.cpp]: in cases like this, put the `>` on the next line and align the `parse` with the previous line

File: parse/ValueRefParserImpl.h [hunk 11]
```diff
@@ -102,8 +102,9 @@ void initialize_nonnumeric_statistic_parser(
     const parse::lexer& tok = parse::lexer::instance();
 
     statistic
-        =   tok.Statistic_
-            >>  tok.Mode_ [ _b = ValueRef::MODE ]
+        =   ((tok.Statistic_
+            >>  tok.Mode_ [ _b = ValueRef::MODE ])
```
Comment (geoffthemedio) [inline parse/ValueRefParserImpl.h]: ...

**Review by LGM-Doyle — COMMENTED** [review]
Once again, all fixed.

**Comment by geoffthemedio** [comment]
Still not really happy with the code layout, but I'm not sure if there's a great solution for me given the positioning requirements of the brackets for preventing this issue... so OK as is, I guess.

**Pull request merged**